### ISSUE TITLE
SQL function not working when trying to write table to non-default schema

### ISSUE THREAD

**Issue by jrisi256** [r-dbi/odbc#265]
I am trying to write a table to a non-default schema in SQL Server 2017. I am using RStudio's documentation as to what the best practice is for doing this: https://db.rstudio.com/best-practices/schema/#write-non-temporary-tables.

However, when I follow their guidelines I get the following error: `Error: Can't unquote EXISTING_SCHEMA.newTbleIris`

### Reproducible Example
```r
library(odbc)
library(dbplyr)
library(dplyr)
library(DBI)

con <- dbConnect(odbc(),
                 Driver = "SQL SERVER",
                 Server = "SERVER_NAME",
                 Database = "DB_NAME",
                 UID = "USER_NAME",
                 PWD = "PASSWORD")

# this works, writing to default schema
dbWriteTable(con, SQL("newTbleIris"), iris)

# this does NOT work and gives error mentioned above
dbWriteTable(con, SQL("EXISTING_SCHEMA.newTbleIris"), iris)
```
### Session Info
```r
- Session info --------------------------------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.5.2 (2018-12-20)
 os       Windows Server >= 2012 x64  
 system   x86_64, mingw32             
 ui       RStudio                     
 language (EN)                        
 collate  English_United States.1252  
 ctype    English_United States.1252  
 tz       America/New_York            
 date     2019-03-07                  

- Packages ------------------------------------------------------------------------------------------------------------------------
 package     * version date       lib source        
 assertthat    0.2.0   2017-04-11 [1] CRAN (R 3.5.2)
 backports     1.1.3   2018-12-14 [1] CRAN (R 3.5.2)
 bit           1.1-14  2018-05-29 [1] CRAN (R 3.5.2)
 bit64         0.9-7   2017-05-08 [1] CRAN (R 3.5.2)
 blob          1.1.1   2018-03-25 [1] CRAN (R 3.5.2)
 callr         3.1.1   2018-12-21 [1] CRAN (R 3.5.2)
 cli           1.0.1   2018-09-25 [1] CRAN (R 3.5.2)
 crayon        1.3.4   2017-09-16 [1] CRAN (R 3.5.2)
 DBI         * 1.0.0   2018-05-02 [1] CRAN (R 3.5.2)
 dbplyr      * 1.3.0   2019-01-09 [1] CRAN (R 3.5.2)
 desc          1.2.0   2018-05-01 [1] CRAN (R 3.5.2)
 devtools      2.0.1   2018-10-26 [1] CRAN (R 3.5.2)
 digest        0.6.18  2018-10-10 [1] CRAN (R 3.5.2)
 dplyr       * 0.8.0.1 2019-02-15 [1] CRAN (R 3.5.2)
 fs            1.2.6   2018-08-23 [1] CRAN (R 3.5.2)
 glue          1.3.0   2018-07-17 [1] CRAN (R 3.5.2)
 hms           0.4.2   2018-03-10 [1] CRAN (R 3.5.2)
 htmltools     0.3.6   2017-04-28 [1] CRAN (R 3.5.2)
 httpuv        1.4.5.1 2018-12-18 [1] CRAN (R 3.5.2)
 jsonlite      1.6     2018-12-07 [1] CRAN (R 3.5.2)
 later         0.8.0   2019-02-11 [1] CRAN (R 3.5.2)
 magrittr      1.5     2014-11-22 [1] CRAN (R 3.5.2)
 memoise       1.1.0   2017-04-21 [1] CRAN (R 3.5.2)
 mime          0.6     2018-10-05 [1] CRAN (R 3.5.2)
 odbc        * 1.1.6   2018-06-09 [1] CRAN (R 3.5.2)
 pillar        1.3.1   2018-12-15 [1] CRAN (R 3.5.2)
 pkgbuild      1.0.2   2018-10-16 [1] CRAN (R 3.5.2)
 pkgconfig     2.0.2   2018-08-16 [1] CRAN (R 3.5.2)
 pkgload       1.0.2   2018-10-29 [1] CRAN (R 3.5.2)
 prettyunits   1.0.2   2015-07-13 [1] CRAN (R 3.5.2)
 processx      3.2.1   2018-12-05 [1] CRAN (R 3.5.2)
 promises      1.0.1   2018-04-13 [1] CRAN (R 3.5.2)
 ps            1.3.0   2018-12-21 [1] CRAN (R 3.5.2)
 purrr       * 0.3.0   2019-01-27 [1] CRAN (R 3.5.2)
 R6            2.4.0   2019-02-14 [1] CRAN (R 3.5.2)
 Rcpp          1.0.0   2018-11-07 [1] CRAN (R 3.5.2)
 remotes       2.0.2   2018-10-30 [1] CRAN (R 3.5.2)
 rlang         0.3.1   2019-01-08 [1] CRAN (R 3.5.2)
 rprojroot     1.3-2   2018-01-03 [1] CRAN (R 3.5.2)
 rstudioapi    0.9.0   2019-01-09 [1] CRAN (R 3.5.2)
 sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 3.5.2)
 shiny       * 1.2.0   2018-11-02 [1] CRAN (R 3.5.2)
 tibble        2.0.1   2019-01-12 [1] CRAN (R 3.5.2)
 tidyselect    0.2.5   2018-10-11 [1] CRAN (R 3.5.2)
 usethis       1.4.0   2018-08-14 [1] CRAN (R 3.5.2)
 withr         2.1.2   2018-03-15 [1] CRAN (R 3.5.2)
 xtable        1.8-3   2018-08-29 [1] CRAN (R 3.5.2)
```

**Comment by mkirzon** [comment]
Are there any proposed workarounds for this in the meantime?

**Comment by jimhester** [comment]
This seems to be working fine now, for both creating, appending and querying.

``` r
library(DBI); con <- dbConnect(odbc::odbc(), "MicrosoftSQLServer", UID = 'testuser', PWD = 'passW0rd!', database = 'test')

dbExecute(con, "CREATE SCHEMA testSchema")
#> [1] 0

tbl_id <- Id(schema = "testSchema", table = "newTableIris")
dbWriteTable(con, "newTableIris", iris)
dbWriteTable(con, tbl_id, iris)
dbAppendTable(con, tbl_id, iris)
#> [1] 150
nrow(dbReadTable(con, "newTableIris"))
#> [1] 150
nrow(dbReadTable(con, tbl_id))
#> [1] 300
dbRemoveTable(con, tbl_id)
dbRemoveTable(con, "newTableIris")

dbExecute(con, "DROP SCHEMA testSchema")
#> [1] 0
```

<sup>Created on 2019-11-15 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

@edgararuiz, I do think we should update the referenced documentation to use `DBI::Id()`, the usage of `SQL()` for this won't work as written. If we do want to show `SQL()` you need to have the example include the quotes around the identifiers, e.g. `SQL('"testSchema"."newTableIris"')`

**Comment by jrisi256** [comment]
Thank you @jimhester ! Is this in latest release of package? Or is there a specific commit this was fixed in? I will test later to make sure this works for me.

**Comment by zoews** [comment]
hey @jimhester ! i found this thread and the `Id()` method of defining schema was the only way I could successfully pass along a table name in SQL Server - lots of other stuff failed. 

Is there a way to highlight this more clearly in the docs? in particular, when I looked at the doc for `DBI::dbWriteTable` I found this for the `name` argument:

```
name	
The table name, passed on to dbQuoteIdentifier(). Options are:

a character string with the unquoted DBMS table name, e.g. "table_name",
a call to Id() with components to the fully qualified table name, e.g. Id(schema = "my_schema", table = "table_name")
a call to SQL() with the quoted and fully qualified table name given verbatim, e.g. SQL('"my_schema"."table_name"')
```
Perhaps it's worth clarifying that character string will silently fail if you are trying to pass in [schema].[table]? (or include a validation looking for ".") The issue seems obvious in retrospect but it might help minimize future DB integration flailing from ppl like me haha. (**Happy to work on a PR for this if that'd be useful!**) Thank you!
